src/logic/useBikeRentalStations.ts

```diff
@@ -1,61 +1,87 @@
-import { BikeRentalStation } from '@entur/sdk'
+import { useState, useEffect } from 'react'
+import { Coordinates } from '@entur/sdk'
+import { Station } from '@entur/sdk/lib/mobility/types'
 
-import { usePrevious, isNotNullOrUndefined } from '../utils'
 import service from '../service'
 import { useSettingsContext } from '../settings'
-import { REFRESH_INTERVAL } from '../constants'
+import { includesStation } from '../utils'
 
-import useNearestPlaces from './useNearestPlaces'
-
-async function fetchBikeRentalStations(
+async function fetchBikeRentalStationsById(
     allStationIds: string[],
-): Promise<BikeRentalStation[] | null> {
-    const allStations = await service.getBikeRentalStations(allStationIds)
-    return allStations.filter(isNotNullOrUndefined)
+): Promise<Station[] | null> {
+    const allStations = await service.mobility.getStationsById({
+        stationIds: allStationIds,
+    })
+    return allStations
+}
+
+async function fetchBikeRentalStationsNearby(
+    coordinates: Coordinates,
+    distance: number,
+): Promise<Station[] | null> {
+    const allStations = await service.mobility.getStations({
+        lat: coordinates.latitude,
+        lon: coordinates.longitude,
+        range: distance,
+    })
+    return allStations
 }
 
-export default function useBikeRentalStations(): BikeRentalStation[] | null {
+export default function useBikeRentalStations(
+    removeHiddenStations = true,
+): Station[] | null {
     const [settings] = useSettingsContext()
     const [bikeRentalStations, setBikeRentalStations] = useState<
-        BikeRentalStation[] | null
+        Station[] | null
     >(null)
-    const nearestPlaces = useNearestPlaces(
-        settings?.coordinates,
-        settings?.distance,
+    const [nearbyStations, setNearbyStations] = useState<Station[]>([])
+    const [userSelectedStations, setUserSelectedStations] = useState<Station[]>(
+        [],
     )
 
-    const { newStations = [], hiddenStations, hiddenModes } = settings || {}
-
-    const nearestBikeRentalStations = useMemo(
-        () =>
-            nearestPlaces
-                .filter(({ type }) => type === 'BikeRentalStation')
-                .map(({ id }) => id),
-        [nearestPlaces],
-    )
+    const {
+        coordinates,
+        distance,
+        newStations = [],
+        hiddenStations = [],
+        hiddenModes,
+    } = settings || {}
 
-    const allStationIds = [...newStations, ...nearestBikeRentalStations]
-        .filter((id) => !hiddenStations?.includes(id))
-        .filter((id, index, ids) => ids.indexOf(id) === index)
+    const isDisabled = Boolean(hiddenModes?.includes('bysykkel'))
 
-    const prevStationIds = usePrevious(allStationIds)
+    useEffect(() => {
+        if (!coordinates || !distance || isDisabled) {
+            return setBikeRentalStations(null)
+        }
+        fetchBikeRentalStationsNearby(coordinates, distance).then((stations) =>
+            setNearbyStations(stations || []),
```

Comment:
I desse useEffectene går det an å bruke AbortController sånn som det er gjort ein del andre plassar.

Reply:
Ja, tenkte på det. Ville bare ikke gjøre for mye ekstra refaktorering, så PR'en ble unødvendig stor. Men kan fikse det etterpå, så får vi ryddet opp litt mer.

Reply:
La det inn i en commit her.